### PR TITLE
fix(react-avatar): Remove warning from AvatarGroup and fix problem with fragments in AvatarGroup

### PR DESCRIPTION
**Pull request by sopranopillow**
## Current Behavior

AvatarGroup warns when an Avatar component is used inside an AvatarGroup.

When given a fragment containing the AvatarGroupItems such as:
```
<AvatarGroup>
  <>
    <AvatarGroupItem />
    <AvatarGroupItem />
    <AvatarGroupItem />
    <AvatarGroupItem />
    <AvatarGroupItem />
  </>
</AvatarGroup>
```

AvatarGroup does not handle the items correctly and overflow is not handled correctly.

## New Behavior

As mentioned by Shift in this [comment](https://github.com/microsoft/fluentui/pull/23498#issuecomment-1161447533), AvatarGroup's warning is breaking the composition pattern. To avoid this, the warning has been removed.

AvatarGroup now checks if the children contains a fragment and gets its contents.
Note: this is a temporary fix of a bigger problem, to follow up on this issue, see #23666.

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/Avatar-mount_base.html">1372</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/Avatar-mount.html">1373</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/Button-mount_base.html">1018</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/Button-mount.html">989</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProvider-mount_base.html">2339</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProvider-mount.html">1895</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-mount_base.html">758</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-mount.html">696</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">706</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">778</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">739</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/MakeStyles-mount_base.html">2045</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test-react-components/MakeStyles-mount.html">2058</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/BaseButton-mount_base.html">757</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/BaseButton-mount.html">793</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Breadcrumb-mount_base.html">2368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Breadcrumb-mount.html">2373</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Checkbox-mount_base.html">1305</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Checkbox-mount.html">1264</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/CheckboxBase-mount_base.html">1069</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/CheckboxBase-mount.html">1104</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ChoiceGroup-mount_base.html">4165</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ChoiceGroup-mount.html">4126</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ComboBox-mount_base.html">838</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ComboBox-mount.html">868</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/CommandBar-mount_base.html">9063</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/CommandBar-mount.html">9038</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ContextualMenu-mount_base.html">10103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ContextualMenu-mount.html">10006</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DefaultButton-mount_base.html">971</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DefaultButton-mount.html">965</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRow-mount_base.html">3297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRow-mount.html">3281</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRowFast-mount_base.html">3334</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRowFast-mount.html">3307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRowNoStyles-mount_base.html">3212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DetailsRowNoStyles-mount.html">3130</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Dialog-mount_base.html">2473</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Dialog-mount.html">2474</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DocumentCardTitle-mount_base.html">152</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/DocumentCardTitle-mount.html">150</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Dropdown-mount_base.html">3139</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Dropdown-mount.html">2900</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/FocusTrapZone-mount_base.html">1667</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/FocusTrapZone-mount.html">1686</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/FocusZone-mount_base.html">1556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/FocusZone-mount.html">1600</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/IconButton-mount_base.html">1519</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/IconButton-mount.html">1504</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Label-mount_base.html">299</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Label-mount.html">315</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Layer-mount_base.html">2520</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Layer-mount.html">2544</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Link-mount_base.html">398</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Link-mount.html">405</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/MenuButton-mount_base.html">1269</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/MenuButton-mount.html">1253</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/MessageBar-mount_base.html">1791</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/MessageBar-mount.html">1896</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Nav-mount_base.html">2841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Nav-mount.html">2858</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/OverflowSet-mount_base.html">950</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/OverflowSet-mount.html">939</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Panel-mount_base.html">1884</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Panel-mount.html">1908</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Persona-mount_base.html">872</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Persona-mount.html">852</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Pivot-mount_base.html">1253</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Pivot-mount.html">1229</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/PrimaryButton-mount_base.html">1126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/PrimaryButton-mount.html">1118</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Rating-mount_base.html">6713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Rating-mount.html">6704</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SearchBox-mount_base.html">1108</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SearchBox-mount.html">1120</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Shimmer-mount_base.html">2098</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Shimmer-mount.html">2147</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Slider-mount_base.html">1689</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Slider-mount.html">1664</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SpinButton-mount_base.html">4312</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SpinButton-mount.html">4354</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Spinner-mount_base.html">381</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Spinner-mount.html">371</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SplitButton-mount_base.html">2737</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SplitButton-mount.html">2713</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Stack-mount_base.html">443</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Stack-mount.html">453</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/StackWithIntrinsicChildren-mount_base.html">2023</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/StackWithIntrinsicChildren-mount.html">2009</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/StackWithTextChildren-mount_base.html">4537</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/StackWithTextChildren-mount.html">4513</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SwatchColorPicker-mount_base.html">10053</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/SwatchColorPicker-mount.html">10120</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TagPicker-mount_base.html">2349</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TagPicker-mount.html">2347</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TeachingBubble-mount_base.html">80149</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TeachingBubble-mount.html">80026</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Text-mount_base.html">355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Text-mount.html">363</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TextField-mount_base.html">1208</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/TextField-mount.html">1245</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-mount_base.html">1023</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-mount.html">1024</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-virtual-rerender_base.html">549</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-virtual-rerender.html">550</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1617</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1618</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Toggle-mount_base.html">681</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/Toggle-mount.html">696</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/buttonNative-mount_base.html">111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23628/perf-test/buttonNative-mount.html">108</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

**Comment by ling1726** [comment]
> to follow up on this issue, see

sorry what issue ? 😅

File: yarn.lock [hunk 1]
```diff
@@ -1652,11 +1652,81 @@
     "@uifabric/set-version" "^7.0.23"
     tslib "^1.10.0"
 
+"@fluentui/react-alert@9.0.0-beta.2":
```
Comment (ling1726) [inline yarn.lock:1655]: you should remove the package.json and yarn.lock changes after master is fixed #23634

File: packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItem.ts [hunk 2]
```diff
@@ -26,12 +26,6 @@ export const useAvatarGroupItem_unstable = (
   // Since the primary slot is not an intrinsic element, getPartitionedNativeProps cannot be used here.
   const { style, className, ...avatarSlotProps } = props;
   const size = groupSize ?? defaultAvatarGroupSize;
-  const hasAvatarGroupContext = useHasParentContext(AvatarGroupContext);
-
-  if (process.env.NODE_ENV !== 'production' && !hasAvatarGroupContext) {
```
Comment (ling1726) [inline packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItem.ts:31]: not sure why you removed this warning, this is actually good

File: packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroup.tsx [hunk 3]
```diff
@@ -17,15 +16,9 @@ import type { AvatarGroupProps, AvatarGroupState } from './AvatarGroup.types';
...
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    childrenArray.find(child => React.isValidElement(child) && child.type !== AvatarGroupItem)
-  ) {
-    // eslint-disable-next-line no-console
-    console.warn("AvatarGroup's children must be of type AvatarGroupItems.");
-  }
+  const childrenArray = React.Children.toArray(
+    React.isValidElement(children) && children.type === React.Fragment ? children.props.children : children,
+  );
```
Comment (layershifter) [inline packages/react-components/react-avatar/src/components/AvatarGroup/useAvatarGroup.tsx:21]: Don't want to make it worse, but:

```tsx
<AvatarGroup>
  <>
    <AvatarGroupItem />
    <AvatarGroupItem />
    <>
      <AvatarGroupItem />
      <AvatarGroupItem />
      <AvatarGroupItem />
    </>
  </>
</AvatarGroup>
```

**Pull request closed without merging**